src/System.Net.Http/src/System/Net/Http/Managed/HttpConnection.cs

```diff
@@ -209,13 +204,26 @@ private Task WriteFormattedInt32Async(int value, CancellationToken cancellationT
                     request.Headers.TransferEncodingChunked = true;
                 }
 
+                if (request.Version.Major != 1)
+                {
+                    // TODO #23134: Support 2.0
+                    throw new PlatformNotSupportedException($"Only HTTP 1.0 supported -- request.Version was {request.Version}");
+                }
+                else if (request.Version.Minor == 0 && request.Headers.TransferEncodingChunked == true)
+                {
+                    // HTTP 1.0 does not support chunking
+                    throw new NotSupportedException($"HTTP 1.0 does not support chunking.");
+                }
+
                 // Write request line
                 await WriteStringAsync(request.Method.Method, cancellationToken).ConfigureAwait(false);
                 await WriteByteAsync((byte)' ', cancellationToken).ConfigureAwait(false);
                 await WriteStringAsync(
                     _usingProxy ? request.RequestUri.AbsoluteUri : request.RequestUri.PathAndQuery,
                     cancellationToken).ConfigureAwait(false);
-                await WriteBytesAsync(s_spaceHttp11NewlineAsciiBytes, cancellationToken).ConfigureAwait(false);
+                await WriteBytesAsync(
+                    request.Version.Minor == 1 ? s_spaceHttp11NewlineAsciiBytes : s_spaceHttp10NewlineAsciiBytes,
```

Comment:
If someone specified a version of 1.2, this is going to send 1.0 in the header.  Is that correct?  Or do we want to send 1.2?  Or 1.1?  What do the existing handlers do?  We should have a test for it and match the behavior.

Reply:
>What do the existing handlers do? We should have a test for it and match the behavior.

Desired behavior should be to use 1.1 for all unrecognized versions.

Reply:
I don't know. I'll do some testing. Same for the chunking question above. It would be easy to add check to the version logic above to invalid/unsupported versions.